servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/AbstractCloseableIteratorAsInputStream.java

```diff
@@ -164,9 +164,13 @@ public final int read() throws IOException {
         }
     }
 
-    private void checkAlreadyClosed() throws IOException {
+    protected void checkAlreadyClosed() throws IOException {
         if (isClosed()) {
             throw new IOException("Stream is already closed.");
         }
     }
+
+    protected boolean hasNext() {
```

Comment:
Btw, 2 implementations that we currently have don't hold a reference to `iterator`. Should we pass it there as an argument. I see you had it originally but then removed, why?

Reply:
Actually, turns out, we don't need to override `hasNext` I think so I will clean this change up a bit.